sdgx/data_models/multi_table_combiner.py

```diff
@@ -1,2 +1,54 @@
-class MultiTableCombiner:
-    pass
+from typing import Any, Dict, List
+
+from pydantic import BaseModel
+
+from sdgx.data_models.metadata import Metadata
+from sdgx.data_models.relationship import Relationship
+
+
+class MultiTableCombiner(BaseModel):
+    """MultiTableCombiner: combine different tables using relationship
+
+    Args:
+        metadata_dict (Dict[str, Any]):
+
+        relationships (List[Any]):
+    """
+
+    metadata_version: str = "1.0"
+
+    metadata_dict: Dict[str, Any] = {}
+    relationships: List[Any] = []
+
+    def check(self):
+        """Do necessary checks:
+
+        - Whether number of tables corresponds to relationships.
+        - Whether table names corresponds to the relationship between tables;
+        """
+
+        # count check
+        relationship_cnt = len(self.relationships)
+        metadata_cnt = len(self.metadata_dict.keys())
+        if metadata_cnt != relationship_cnt + 1:
+            raise ValueError("Number of tables should corresponds to relationships.")
```

Comment:
Can we add a new exception in `sdgx/exceptions.py`? This will help us differentiate between what are *expected* exceptions.

Reply:
This is something I didn't consider, I'll add new exceptions `MultiTableCombineError` and `MetadataError` in `sdgx.exceptions`.

Reply:
For better unit testing and bug fixing, I'd suggest synchronizing the main branch, which also includes `Exceptions` with error codes and exit codes